lib/chef/resource/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,46 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class CloudSearch
+    class Domain
+      # The version of the AWS sdk we are using doesn't have a model
+      # object for CloudSearch Domains. This empty class is here to
+      # make the reset of chef-provisioning happy.
+    end
+  end
+end
+
+class Chef::Resource::AwsCloudsearchDomain < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::AWS::CloudSearch::Domain
+  attribute :domain_name, kind_of: String, name_attribute: true
```

Comment:
It enables logic where if you have

```ruby
aws_cloudsearch_domain aws_object do
  ...
end
```

Where `aws_object` is an AWS model that you have gotten from somewhere, then the resource will still converge successfully.  But because cloud search doesn't have an AWS model then I think its okay to leave it off for now.